### DISCUSSION TITLE
Framework for project from scratch

### DISCUSSION THREAD

**Discussion by Remo** [peachpiecompiler/peachpie#1095] [Q&A] [answered]
I was wondering what framework suits peachpie if I wanted to start a project from scratch.

It might sound a bit strange, but I've got a lot of PHP resources, but need some libraries that are only available for .NET.

I know there are ways to call it from Php, but I am still considering to use peachpie.

What would you guys use if you wanted to start a new project with peachpie. It would basically be an API backend that receives and processes Json and a UI built with Vuejs.

I like laravel, but it does not feel like it matches the world of .NET very well. Too many dynamic things, files written on the fly, facades and so on.

Any ideas?

**Comment by jakubmisek** [comment, marked as answer]
Thank you for the question!

We're personally using PeachPie to "revive" legacy custom-made PHP software, turn them into a .NET library, and extend/interoperate with the C# code base.

Anyways;

For the best performance - I'd use ASP.NET Controllers to handle the requests, the rest can be written in PHP.

For the all-PHP-code - Laravel would be an overkill indeed. Maybe choose just the important symfony components? Or we've made the [Slim framework working](https://github.com/insinfo/testeSlimPeachPie) on PeachPie as well.

**Comment by Remo** [comment]
Thanks! I think give slim a chance. We'll need a lot of controllers, adding them in PHP would make it easier for PHP devs to take care of the whole project.
